bundles/org.openhab.binding.icalendar/src/main/java/org/openhab/binding/icalendar/internal/handler/PullJob.java

```diff
@@ -100,8 +102,17 @@ public void run() {
         Response response;
         try {
             response = asyncListener.get(HTTP_TIMEOUT_SECS, TimeUnit.SECONDS);
-        } catch (InterruptedException | TimeoutException | ExecutionException e1) {
-            logger.warn("Response for calendar request could not be retrieved. Error message is: {}", e1.getMessage());
+        } catch (InterruptedException e1) {
+            logger.warn("Download of calendar was interrupted.");
+            logger.debug("InterruptedException message is: {}", e1.getMessage());
+            return;
+        } catch (TimeoutException e1) {
+            logger.warn("Download of calendar timed out (waited too long for headers).");
+            logger.debug("TimeoutException message is: {}", e1.getMessage());
+            return;
+        } catch (ExecutionException e1) {
+            logger.warn("Download of calendar failed while execution.");
+            logger.warn("ExecutionException message is: {}", e1.getMessage());
```

Comment:
An ExecutionException is used to wrap other exceptions, so it never contains any useful message on its own.
It is better to log the cause exception instead.
Also, Is this supposed to be debug as well?
```suggestion
            logger.debug("ExecutionException message is: {}", e1.getCause().getMessage());
```

Reply:
Yes, it's better to log the cause if the Exception itself is not explanatory.  I thought, warn would also fit for this type of exception as it's quite general. But i don't have any problem to move it to debug.